jobs/future-effective-filings/Dockerfile

```diff
@@ -1,35 +1,80 @@
-# platform=linux/amd64
-FROM python:3.8.6-buster
+FROM python:3.11.2-bullseye AS development_build
+
+USER root
 
 ARG VCS_REF="missing"
 ARG BUILD_DATE="missing"
 
 ENV VCS_REF=${VCS_REF}
 ENV BUILD_DATE=${BUILD_DATE}
+ENV PORT=8080
 
 LABEL org.label-schema.vcs-ref=${VCS_REF} \
     org.label-schema.build-date=${BUILD_DATE}
 
-USER root
+LABEL maintainer="karimjazzar"
+LABEL vendor="BCROS"
+
+ARG APP_ENV \
+  # Needed for fixing permissions of files created by Docker:
+  UID=1000 \
+  GID=1000
+
+ENV APP_ENV=${APP_ENV} \
+  # python:
+  PYTHONFAULTHANDLER=1 \
+  PYTHONUNBUFFERED=1 \
+  PYTHONHASHSEED=random \
+  PYTHONDONTWRITEBYTECODE=1 \
+  # pip:
+  PIP_NO_CACHE_DIR=1 \
+  PIP_DISABLE_PIP_VERSION_CHECK=1 \
+  PIP_DEFAULT_TIMEOUT=100 \
+  PIP_ROOT_USER_ACTION=ignore \
+  # poetry:
+  POETRY_VERSION=1.3.2 \
+  POETRY_NO_INTERACTION=1 \
+  POETRY_VIRTUALENVS_CREATE=false \
+  POETRY_CACHE_DIR='/var/cache/pypoetry' \
+  POETRY_HOME='/usr/local'
+
+SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
+
+RUN apt-get update && apt-get upgrade -y \
+  && apt-get install --no-install-recommends -y \
+    bash \
+    brotli \
+    build-essential \
+    curl \
+    gettext \
+    git \
+    libpq-dev \
+    wait-for-it \
+  && curl -sSL 'https://install.python-poetry.org' | python - \
+  && poetry --version \
+  # Cleaning cache:
+  && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+  && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
-# Create working directory
-RUN mkdir /opt/app-root && chmod 755 /opt/app-root
-WORKDIR /opt/app-root
+WORKDIR /code
 
-# Install the requirements
-COPY ./requirements.txt .
+RUN groupadd -g "${GID}" -r web \
+  && useradd -d '/code' -g web -l -r -u "${UID}" web \
+  && chown web:web -R '/code'
 
-#RUN pip install --upgrade pip
-RUN pip install pip==20.1.1
-RUN pip install --no-cache-dir -r requirements.txt
+# Copy only requirements, to cache them in docker layer
+COPY --chown=web:web ./poetry.lock ./pyproject.toml /code/
 
-COPY . .
+COPY --chown=web:web ./src /code/src
+COPY --chown=web:web ./README.md /code
 
-USER 1001
+# Running as non-root user:
+USER web
 
-# Set Python path
-ENV PYTHONPATH=/opt/app-root/src
+# The following stage is only for production:
+FROM development_build AS production_build
+COPY --chown=web:web . /code
 
-EXPOSE 8080
+# ENV PYTHONPATH=/opt/app-root/src
 
-CMD [ "python", "/opt/app-root/file_future_effective.py" ]
+CMD gunicorn --bind 0.0.0.0:${PORT} --config /code/gunicorn_config.py wsgi:app
```

Comment:
I'm don't think we need all the gunicorn stuff.  I think we should just be calling  `/code/file_future_effective.py`.  

Try re-working the original call(`
CMD [ "python", "/opt/app-root/file_future_effective.py" ]`), build the docker container locally and see if you can run the container.  If things work, I think we can get rid of the gunicorn code.

Reply:
Argus, you're right. I changed the call to the one you mentioned and build the docker container using:
`docker build -t future-effective-filings -f Dockerfile .`
![docker build](https://github.com/bcgov/lear/assets/122301442/33fefec2-6080-45d4-8bd7-6461b11afa71)

But whenever I try to run the image I created:
![docker run](https://github.com/bcgov/lear/assets/122301442/68b24af0-9fbc-45a8-8178-2cee0bb0c39e)

The import isn't being resolved although requests is in `pyproject.toml` and I can see it in my IDE.